.circleci/config.yml

```diff
@@ -0,0 +1,181 @@
+version: 2
+
+workflows:
```

Comment:
This is so nice looking, but one reason I didn't really touch Travis here (all the while keeping an eye on GitHub CI), is that I'm a bit afraid of CircleCI's free limits. I believe there's a maximum of 4 jobs for our organization, plus even just for front I've at least once received a warning that we were more or less at some monthly computation time limit.

Reply:
yes, max concurrency for circleci is 4. do you know what's the computation time limit for travis ci?

Reply:
I have no idea, but I've never seen any warnings and it's likely always used up more than CircleCI. :-)

Reply:
Do you have any idea why CircleCI is a few minutes faster btw?

Reply:
OK, we can keep using travis since my android test change should also work for travis ci build.

Given that travis ci has recently been acquired and the new parent company is firing engineers left and right, I think eventually we will have to move off it to something else. I am going to keep the circleci config and just disable circleci build from their UI for now.

I think CircleCI build is faster because they have beefier machine?

Reply:
oh wow, travis ci is very slow. the full circle ci build is almost done. travis ci is still in queuing state and hasn't started even a single job.

Reply:
Ah, so, not just on my PR ;p. It was starting fine à few hours ago, FWIW, but, err, I've kinda been hammering jobs because of those cmake shenanigans that I can't reproduce locally...

Reply:
@houqp Usually Travis takes a little longer to get going but I mean at most tens of seconds.

CircleCI does occasionally also end up with long queues when it's busy.

> I think eventually we will have to move off it to something else.

I'm keeping an eye on GitHub Actions.

https://medium.com/selleo/experimenting-with-github-actions-beta-760e61fab0e1

@NiLuJe CMake 3.6 seemed like a sensible version to support, but we can always go for newer if necessary.